File: invesdwin-context-parent/invesdwin-context-jruby/src/test/java/de/invesdwin/context/jruby/tests/callback/ParametersAndReturnsTestBoolean.rb
puts("getBoolean")
unless (defined?(getBoolean)).nil? || getBoolean.nil?
	raise "getBoolean already defined!"
end
getBoolean = callback("getBoolean")
getBooleanType = getBoolean.class
puts(getBooleanType)
puts(getBoolean)
if(getBooleanType != TrueClass and getBooleanType != FalseClass)
	raise "getBoolean not boolean!"
end
callback("setBoolean", getBoolean)

puts("getBooleanVector")
unless (defined?(getBooleanVector)).nil? || getBooleanVector.nil?
	raise "getBooleanVector already defined!"
end
getBooleanVector = callback("getBooleanVector")
getBooleanVectorType = getBooleanVector[0].class
puts(getBooleanVectorType)
puts(getBooleanVector)
if(getBooleanVectorType != TrueClass and getBooleanVectorType != FalseClass)
	raise "getBooleanVector not boolean!"
end
callback("setBooleanVector", getBooleanVector)

puts("getBooleanVectorAsList")
unless (defined?(getBooleanVectorAsList)).nil? || getBooleanVectorAsList.nil?
	raise "getBooleanVectorAsList already defined!"
end
getBooleanVectorAsList = callback("getBooleanVectorAsList")
getBooleanVectorAsListType = getBooleanVectorAsList[0].class
puts(getBooleanVectorAsListType)
puts(getBooleanVectorAsList)
if(getBooleanVectorAsListType != TrueClass and getBooleanVectorAsListType != FalseClass)
	raise "getBooleanVectorAsList not boolean!"
end
callback("setBooleanVectorAsList", getBooleanVectorAsList)

puts("getBooleanMatrix")
unless (defined?(getBooleanMatrix)).nil? || getBooleanMatrix.nil?
	raise "getBooleanMatrix already defined!"
end
getBooleanMatrix = callback("getBooleanMatrix")
getBooleanMatrixType = getBooleanMatrix[0][0].class
puts(getBooleanMatrixType)
puts(getBooleanMatrix)
if(getBooleanMatrixType != TrueClass and getBooleanMatrixType != FalseClass)
	raise "getBooleanMatrix not boolean!"
end
callback("setBooleanMatrix", getBooleanMatrix)

puts("getBooleanMatrixAsList")
unless (defined?(getBooleanMatrixAsList)).nil? || getBooleanMatrixAsList.nil?
	raise "getBooleanMatrixAsList already defined!"
end
getBooleanMatrixAsList = callback("getBooleanMatrixAsList")
getBooleanMatrixAsListType = getBooleanMatrixAsList[0][0].class
puts(getBooleanMatrixAsListType)
puts(getBooleanMatrixAsList)
if(getBooleanMatrixAsListType != TrueClass and getBooleanMatrixAsListType != FalseClass)
	raise "getBooleanMatrixAsList not boolean!"
end
callback("setBooleanMatrixAsList", getBooleanMatrixAsList)

File: invesdwin-context-parent/invesdwin-context-clojure/src/test/java/de/invesdwin/context/clojure/tests/callback/ParametersAndReturnsTestBoolean.clj
(defmacro defined [& var]
    `(try 
        (or 
            (boolean (resolve ~@var)) 
            (not (nil? (eval ~@var)))
        ) (catch Exception e# false)
    )
)

(println "getBoolean")
(if (defined 'getBoolean)
	(throw (Exception. "getBoolean already defined!"))
)
(def getBoolean (callback "getBoolean"))
(def getBooleanType (class getBoolean))
(println getBooleanType)
(println getBoolean)
(if-not (= getBooleanType Boolean)
	(throw (Exception. "getBoolean not boolean!"))
)
(callback "setBoolean" getBoolean)

(println "getBooleanVector")
(if (defined 'getBooleanVector)
	(throw (Exception. "getBooleanVector already defined!"))
)
(def getBooleanVector (callback "getBooleanVector"))
(def getBooleanVectorType (class (get getBooleanVector 0)))
(println getBooleanVectorType)
(println getBooleanVector)
(if-not (= getBooleanVectorType Boolean)
	(throw (Exception. "getBooleanVector not boolean!"))
)
(callback "setBooleanVector" getBooleanVector)

(println "getBooleanVectorAsList")
(if (defined 'getBooleanVectorAsList)
	(throw (Exception. "getBooleanVectorAsList already defined!"))
)
(def getBooleanVectorAsList (callback "getBooleanVectorAsList"))
(def getBooleanVectorAsListType (class (get getBooleanVectorAsList 0)))
(println getBooleanVectorAsListType)
(println getBooleanVectorAsList)
(if-not (= getBooleanVectorAsListType Boolean)
	(throw (Exception. "getBooleanVectorAsList not boolean!"))
)
(callback "setBooleanVectorAsList" getBooleanVectorAsList)

(println "getBooleanMatrix")
(if (defined 'getBooleanMatrix)
	(throw (Exception. "getBooleanMatrix already defined!"))
)
(def getBooleanMatrix (callback "getBooleanMatrix"))
(def getBooleanMatrixType (class (get (get getBooleanMatrix 0) 0)))
(println getBooleanMatrixType)
(println getBooleanMatrix)
(if-not (= getBooleanMatrixType Boolean)
	(throw (Exception. "getBooleanMatrix not boolean!"))
)
(callback "setBooleanMatrix" getBooleanMatrix)

(println "getBooleanMatrixAsList")
(if (defined 'getBooleanMatrixAsList)
	(throw (Exception. "getBooleanMatrixAsList already defined!"))
)
(def getBooleanMatrixAsList (callback "getBooleanMatrixAsList"))
(def getBooleanMatrixAsListType (class (get (get getBooleanMatrixAsList 0) 0)))
(println getBooleanMatrixAsListType)
(println getBooleanMatrixAsList)
(if-not (= getBooleanMatrixAsListType Boolean)
	(throw (Exception. "getBooleanMatrixAsList not boolean!"))
)
(callback "setBooleanMatrixAsList" getBooleanMatrixAsList)
